analyzere/extras/visualizations.py

```diff
@@ -0,0 +1,272 @@
+                                     style='filled',
+                                     color='lightgrey')
+                    self._graph.node(ls_name)
+                    self._graph.edge(ls_name, name)
+                    self._edges.add((ls_name, name))
+        return name
+
+    def _generate(self, lv_id, with_terms=True, fmt='pdf', rankdir='BT'):
+
+        # reset the internals for regeneration
+        self._sequence.reset()
+        self._unique_nodes.clear()
+        self._edges.clear()
+
+        # get the LayerView by id
+        lv = LayerView.retrieve(lv_id)
+
+        # build filename with format '<lv_id><-with_terms>-<rankdir>.<fmt>'
+        self._filename = lv.id
+        self._filename += '-with_terms' if with_terms else ''
+        self._filename += '-{}'.format(rankdir)
+
+        self._graph.graph_attr['rankdir'] = rankdir
+
+        # potentially update the format
+        self._graph.format = fmt
+
+        # now build the "tree" of nodes
+        self._get_tree(lv.layer, with_terms)
+
+    def view(self, lv_id, with_terms=True, fmt='pdf', rankdir='BT'):
+        """View the generated graph
+
+        Optional parameters that control the formatting of the graph:
+
+           with_terms   specify if the Layer terms should be included in each
+                        node of the graph
+
+           fmt          exposes the graphvis 'format' options which include
+                        'pdf', 'png', etc.
+
+           rankdir      exposes graphvis 'rankdir' option that controls the
+                        orientation of the graph.  Options include
+                        'TB', 'LR', 'BT', 'RL', corresponding to directed
+                        graphs drawn from top to bottom, from left to right,
+                        from bottom to top, and from right to left,
+                        respectively.
+        """
+        self._generate(lv_id, with_terms, fmt, rankdir)
+        self._graph.render(self._filename, view=True)
+
+    def download(self, lv_id, with_terms=True, fmt='pdf', rankdir='BT'):
+        """Obtain a link to downlaod the generated graph
+
+        Optional parameters that control the formatting of the graph:
+
+           with_terms   specify if the Layer terms should be included in each
+                        node of the graph
+
+           fmt          exposes the graphvis 'format' options which include
+                        'pdf', 'png', etc.
+
+           rankdir      exposes graphvis 'rankdir' option that controls the
+                        orientation of the graph.  Options include
+                        'TB', 'LR', 'BT', 'RL', corresponding to directed
+                        graphs drawn from top to bottom, from left to right,
+                        from bottom to top, and from right to left,
+                        respectively.
+        """
+        self._generate(lv_id, with_terms, fmt, rankdir)
+        return FileLink(self._graph.render(self._filename, view=False))
```

Comment:
I believe the `FileLink` object should only be return when the execution is done within an IPython environment. In cases this is not an IPython environment, I would suggest to just return the path to the file.